authenticated.go

```diff
@@ -13,23 +14,25 @@ func authenticated(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "httpPassword not set", http.StatusInternalServerError)
 		return
 	}
-	s := strings.SplitN(r.Header.Get("Authorization"), " ", 2)
-	if len(s) != 2 || s[0] != "Basic" {
+	kind, encoded, found := strings.Cut(r.Header.Get("Authorization"), " ")
```

Comment:
Can you leave this as-is for now and leave a comment to convert it to strings.Cut instead when Go 1.19 is out?

We generally support the current stable release and (best-effort) the previous stable release, which is Go 1.17 in this case.

Reply:
Sure, I will revert this.

If you want to support go 1.17, you should change the following line in `go.mod`:
https://github.com/gokrazy/gokrazy/blob/4c1cf82b51a18e58718dd34de5d56d3c205b8547/go.mod#L3

https://go.dev/doc/modules/gomod-ref
> The minimum version of Go required by the current module.

Reply:
Can you change it while you’re touching this?

Reply:
Done (I simply copied the original function in the meantime ;)